File: src/completed/python/euler0006.py
###
#
# Created by Holden on 12/17/2015
#
# SOLVED on 12/17/2015
#
# Problem 2 - Sum Square Difference
#
###

def sum_square_difference(ubound):
    sum_of_square = 0
    square_of_sum = 0
    
    for i in range(ubound):
        sum_of_square += (i + 1) ** 2
        
    for i in range(ubound):
        square_of_sum += (i + 1)
        
    square_of_sum = square_of_sum ** 2
    
    return square_of_sum - sum_of_square
    

File: src/completed/java/euler0006.java
/**
 * Created by Holden on 11/20/2016. - COMPLETED
 *
 * Problem 6 - Sum Square Difference
 *
 */

public class Prob0006 {
    public static void main(String[] args) {
        long startTime = System.nanoTime();
        int answer = sumSquareDifference(100);
        long endTime = System.nanoTime();

        System.out.println(answer);

        long duration = (endTime - startTime);
        EulerTime.time(duration);
    }

    public static int sumSquareDifference(int upperBound) {
        long squareofSums = 0;
        for (int i = 1; i <= upperBound; i++) {
            squareofSums += i;
        }
        squareofSums = EulerUtil.pow((int) squareofSums, 2);

        int sumOfSquares = 0;
        for (int i = 1; i <= upperBound; i++) {
            sumOfSquares += EulerUtil.pow(i, 2);
        }

        return (int) (squareofSums - sumOfSquares);
    }
}
